### PR TITLE
[SER-479] Rename tp_client crate to constellation

### PR DESCRIPTION
**Pull request by tomthecarrot**
This PR renames the `tp_client` crate to `constellation` so that:
1. The naming convention is clear to consumers of Constellation's C# library.
2. CppSharp uses the same native library name for both Constellation's C# library and the consumer's C# library (in this case, `unity_states`). As a result, `DllImport` attributes always target the `constellation` library. While the rename isn't required for this purpose, it reduces one step in the build process: renaming `libtp_client.dylib` to `libconstellation.dylib`.

### CODE REVIEW

File: client/build_mac_host.sh [hunk 1]
```diff
@@ -17,7 +17,7 @@ cargo run -p cs_codegen -- -f
 cargo run -p rsharp_codegen -- -f
 
 # Rename library from specific -> generic for symbol compatibility.
-# mv $PLATFORM_DIR/target/debug/libunity_states.dylib $PLATFORM_DIR/target/debug/libconstellation.dylib
+mv $PLATFORM_DIR/target/debug/libunity_states.dylib $PLATFORM_DIR/target/debug/libconstellation.dylib
```
Comment (tomthecarrot) [inline client/build_mac_host.sh:20]: This is a temporary workaround to build and run the `unity_states` demo until we have a CI solution.

**Comment by TheButlah** [comment]
I'm a bit confused as to what the purpose of this PR is. It sounds like the motivation is because `cppsharp` is using `libconstellation.dylib` as the target library, but seems like circular logic because it uses `unity_states`, *not* `libconstellation` currently. Whats to stop us from having cppsharp continue to target `unity_states`?

If the concern is "downstream clients' code won't be called unity states", well they wont be called `constellation` either right? If I am a user and I write a contract, I'll name my contract crate `my_contract_crate` and then write a cppsharp script that uses the `Codegen` class and point it to `libmy_contract_crate.dylib`

Less important (bikeshedding): `tp_client` is the engine, not a client. So it makes some sense to call it `constellation_engine` and there will be `constellation_client` `constellation_server` etc. But do we really want such long import names? Tempted to just call it `engine`, `client`, etc

**Comment by tomthecarrot** [comment]
> Whats to stop us from having cppsharp continue to target `unity_states`?

I thought about this again and tbh the name doesn't matter now that we have all symbols combined into one native library that is generated from the leaf crate of the dependency graph.

My prior reasoning was based on having symbols separated across crates: `DllImport` attributes generated from our PInvoke codegen would be targeting different native libraries (`libtp_client.so`, `librsharp.so`, `libunity_states.so`, etc) from the corresponding managed DLLs (`client.dll`, `rsharp.dll`, etc). Therefore the entrypoints would not be consistent when invoked from different parts of the program: calling a `tp_client` DLL function with the same name as another function in the `unity_states` DLL would invoke a different native entry point.

This is no longer a concern and as such the current PR should likely be closed.

> Less important (bikeshedding): `tp_client` is the engine, not a client. So it makes some sense to call it `constellation_engine` and there will be `constellation_client` `constellation_server` etc. But do we really want such long import names? Tempted to just call it `engine`, `client`, etc

True, and Constellation is really the name of the whole project at Teleportal, not this particular `engine` crate within the project. I'm cool with `engine`.

**Comment by TheButlah** [comment]
I think what we really need rn is for us to be able to control the DLLImport names of the transitive dependencies of the client's contract crate. So we need to be able to reconfigure tp_client and rsharp to use `my_contracts`

This totally goes against the grain of how dependencies should work (dependency code shouldn't change with downstream users names) but it's all we have rn untill we can avoid the "single leaf dylib crate" problem

**Comment by tomthecarrot** [comment]
Closing as unnecessary.

**Pull request closed without merging**